aht20/aht20.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2024 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package aht20
+
+import (
+	"fmt"
+	"periph.io/x/conn/v3/i2c"
+	"periph.io/x/conn/v3/physic"
+	"sync"
+	"time"
+)
+
+const deviceAddress = 0x38
+
+const (
+	cmdStatus     byte = 0x71
+	cmdInitialize byte = 0xBE
+	cmdMeasure    byte = 0xAC
+	cmdSoftReset  byte = 0xBA
+)
+
+const (
+	bitBusy        byte = 1 << 7
+	bitInitialized byte = 1 << 3
+)
+
+var (
+	argsInitialize = []byte{cmdInitialize, 0x08, 0x00}
+	argsMeasure    = []byte{cmdMeasure, 0x33, 0x00}
+)
+
+const crc8Polynomial = uint8(0b00110001) // p(x) = x^8 + x^5 + x^4 + 1. x^8 is omitted due to byte size
+
+type Dev struct {
+	opts Opts
+	d    *i2c.Dev
+	mu   sync.Mutex
+	stop chan struct{}
+	wg   sync.WaitGroup
+}
+
+// Opts holds the configuration options for the device.
+type Opts struct {
+	// MeasurementReadTimeout is the timeout for reading a single measurement. The timeout only applies after the measurement triggering which itself takes 80ms. Default is 150ms. 0 means no timeout.
+	MeasurementReadTimeout time.Duration
+	// MeasurementWaitInterval is the interval between subsequent sensor value reads. This applies only if the measurement is not finished after the initial 80ms wait. Do not confuse this interval with SenseContinuous. Default is 10ms. Leave 0 to use default.
+	MeasurementWaitInterval time.Duration
+	// ValidateData enables data validation using CRC8. If enabled, the sensor will return an error if the data is corrupt. Default is true.
+	ValidateData bool
+}
+
+// DefaultOpts holds the default configuration options for the device.
+var DefaultOpts = Opts{
+	MeasurementReadTimeout:  150 * time.Millisecond,
+	MeasurementWaitInterval: 10 * time.Millisecond,
+	ValidateData:            true,
+}
+
+// NewI2C returns an object that communicates over I²C to AHT20 environmental sensor. The sensor
```

Comment:
It's okay to call it `New` since it's I²C only. The NewI2C naming is generally used when there's also a SPI version.